afs/afs-ws/afs-ws-storage/src/main/java/com/powsybl/afs/ws/storage/NodeEventClient.java

```diff
@@ -54,6 +70,57 @@ public void onError(Throwable t) {
 
     @OnClose
     public void onClose(Session session) {
-        LOGGER.trace("Node event websocket session '{}' closed for file system '{}'", session.getId(), fileSystemName);
+        int counter = 0;
+        boolean toReconnection = true;
+        long dateInitial = Instant.now().getEpochSecond();
+        while (toReconnection) {
+
+            long dateReconnection = Instant.now().getEpochSecond();
+            if (dateReconnection - dateInitial > config.getReconnectionMax()) {
+                LOGGER.warn("Node event websocket session '{}' closed for file system '{}'", session.getId(), fileSystemName);
+                return;
+            }
+            updateReconnectionInterval(counter);
+            try {
+                URI wsUri = RemoteListenableAppStorage.getWebSocketUri(config.getRestUri());
+                URI endPointUri = URI.create(wsUri + "/messages/" + AfsRestApi.RESOURCE_ROOT + "/" +
+                        AfsRestApi.VERSION + "/node_events/" + fileSystemName);
+                LOGGER.debug("Connecting to node event websocket at {}", endPointUri);
+                WebSocketContainer container = ContainerProvider.getWebSocketContainer();
+                container.connectToServer(this, endPointUri);
+                toReconnection = false;
+
+            } catch (DeploymentException e) {
+                if (e.getCause().getCause() instanceof ConnectException) {
+                    try {
+                        LOGGER.error(e.getCause().getCause().getMessage(), e);
+                        counter = counter + 1;
+                        sleep(this.reconnectionInterval * 1000L);
+                    } catch (InterruptedException e1) {
+                        Thread.currentThread().interrupt();
+                        throw new UncheckedInterruptedException(e1);
+                    }
+
+                } else {
+                    LOGGER.warn("Node event websocket session '{}' closed for file system '{}'", session.getId(), fileSystemName);
+                    return;
+                }
+            } catch (IOException e) {
+                throw new UncheckedIOException(e);
+            }
+        }
     }
+
+    private void updateReconnectionInterval(int counter) {
+        if (counter > 0) {
+            if (this.reconnectionInterval > config.getReconnectionMaxInterval()) {
+                this.reconnectionInterval = config.getReconnectionMaxInterval();
+            } else {
+                this.reconnectionInterval = this.reconnectionInterval * config.getReconnectionIntervalMutiplier();
```

Comment:
This may exceed the ReconnectionMaxInterval. We should check against ReconnectionMaxInterval after the multiplication.

Reply:
It's done